### PR TITLE
Chapter_9

### PR DESCRIPTION
**Pull request by tienhung79**
1. Excel: https://docs.google.com/spreadsheets/d/1-YRN-jXuwsdCXTtGjcx9ioq-BpB1SR2PSh7FNml1sp0/edit?usp=sharing
2. Rubocop: 
Screenshot from 2023-08-31 08-27-19

### CODE REVIEW

**Comment by tienhung79** [comment]
Ready

File: app/controllers/session_controller.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class SessionController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by email: params.dig(:session, :email)&.downcase
+    if user&.authenticate params.dig(:session, :password)
+      log_in user
+      params.dig(:session, :remember_me) == "1" ? remember(user) : forget(user)
+      redirect_to user
+
```
Comment (ngocvt-0484) [inline app/controllers/session_controller.rb]: bỏ dòng trống nhé e

File: app/helpers/session_helper.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  def forget user
+    user.forget
+    cookies.delete(:user_id)
```
Comment (ngocvt-0484) [inline app/helpers/session_helper.rb]:
```suggestion
    cookies.delete :user_id
```
hạn chế dùng ngoặc nè

**Comment by tienhung79** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**